docs/architecture/authentication-and-authorization.md

```diff
@@ -0,0 +1,63 @@
+# Authentication and Authorization
+
+## Introduction
+Currently, communication between the Compass and both runtimes and applications are not secured. We need to provide some security possibilities.
+We want to secure the Compass using ORY's Hydra and OathKeeper. There would be two ways of authentication:
+ - Oauth 2.0 
+ - SSL/TLS Certificates.
+
+## Proposed solution
+As mentioned, there would be two possibilities for securing the connection. To achieve that, first, we need to integrate Hydra and OathKeeper into the Compass. Then we will introduce a new component - Tenant Service.
+Tenant Service would be a part of Management Plane Services, responsible for extracting from database a `tenant` based on `client_id`.
+We would also use a Hydrator and ID_token mutator which are ORY's mutators.
```

Comment:
Rephrase this to describe the ID token mutator use case in our implementation. Remove the Hydrator from here, it should be mentioned in the previous sentence.